contributors/design-proposals/daemonset-update.md

```diff
@@ -0,0 +1,222 @@
+  this DaemonSet pod?)
+
+Here are some potential requirements that haven't been covered by this proposal:
+
+- Users can have rate-limited DaemonSet updates (e.g. only upgrade 10% per hour)
+- DaemonSet should provide at-most-one guarantee per node (i.e. at most one pod
+  from a DaemonSet can exist on a node at any time)
+- Uptime is critical for each pod of a DaemonSet during upgrade (e.g. the time
+  from a DaemonSet pods being killed to recreated and healthy should be < 5s)
+- Each DaemonSet pod can still fit on the node after being updated
+- Some DaemonSets require the node to be drained before the DeamonSet's pod on it 
+  is updated (e.g. logging daemons)
+- DaemonSet's pods are implicitly given higher priority than non-daemons
+- DaemonSets can only be operated by admins (i.e. people who manage nodes)
+  - This is required if we allow DaemonSet controllers to drain, cordon,
+    uncordon nodes, evict pods, or allow DaemonSet pods to have higher priority
+
+## Implementation 
+
+### API Object 
+
+To enable DaemonSet upgrades, `DaemonSet` related API object will have the following
+changes:
+
+```go 
+type DaemonSetUpdateStrategy struct {
+	// Type of daemon set update. Can be "RollingUpdate" or "OnDelete". 
+	// Default is OnDelete.
+	Type DaemonSetUpdateStrategyType
+
+	// Rolling update config params. Present only if DaemonSetUpdateStrategy =
+	// RollingUpdate.
+	//---
+	// TODO: Update this to follow our convention for oneOf, whatever we decide it
+	// to be. Same as DeploymentStrategy.RollingUpdate.
+	// See https://github.com/kubernetes/kubernetes/issues/35345
+	RollingUpdate *RollingUpdateDaemonSet
+}
+
+type DaemonSetUpdateStrategyType string
+
+const (
+	// Replace the old daemons by new ones using rolling update i.e replace them on each node one after the other.
+	RollingUpdateDaemonSetStrategyType DaemonSetUpdateStrategyType = "RollingUpdate"
+        
+	// Replace the old daemons only when it's killed
+	OnDeleteDaemonSetStrategyType DaemonSetUpdateStrategyType = "OnDelete"
+)
+
+// Spec to control the desired behavior of daemon set rolling update.
+type RollingUpdateDaemonSet struct {
+	// The maximum number of DaemonSet pods that can be updated at a time.
+	// Value can be an absolute number (ex: 5) or a percentage of total
+	// number of DaemonSet pods at the start of the update (ex: 10%). Absolute
+	// number is calculated from percentage by rounding up.
+	// This cannot be 0.
+	// Default value is 1.
+	// Example: when this is set to 30%, 30% of the currently running DaemonSet
+	// pods can be updated at any given time. The update starts
+	// by stopping at most 30% of the currently running DaemonSet pods and then
+	// brings up new DaemonSet pods in their place. Once the new pods are ready,
+	// it then proceeds onto other DaemonSet pods, thus ensuring that at least
+	// 70% of original number of DaemonSet pods are available at all times
+	// during the update.
+	MaxInFlight intstr.IntOrString
```

Comment:
I think the API bar is to start with Jobs and Deployments and any new concept needs a special argument (in this section) about why it is new.